src/components/common/ClusteringAddPopup.jsx

```diff
@@ -0,0 +1,271 @@
+import React, { Component } from 'react';
+import {
+    Icon,
+    Popup,
+    Button,
+    Modal,
+    Divider,
+    Header,
+    Card,
+    CardContent,
+    Form,
+    Input,
+} from 'semantic-ui-react';
+import { BackendAPI } from './API';
+import _ from 'lodash';
+
+import CSVReader from 'react-csv-reader';
+
+export default class ClusteringAddPopup extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            status: 'ready',
+            showModal: false,
+            clusteringName: '',
+        };
+    }
+
+    closeModal = () => {
+        this.setState({ showModal: false });
+        this.props.handleModalClose();
+    };
+
+    handleModalOpen = () => {
+        this.props.handleModalOpen();
+    };
+
+    onClusteringNameChange = (e) => {
+        this.setState({ clusteringName: e.target.value });
+    };
+
+    validateClusterData = (data, fileInfo) => {
+        console.log(data, fileInfo);
+        let cols = new Set();
+        data.map((x) => {
+            cols.add(x.length);
+        });
+
+        cols.delete(0);
+        if (cols.size != 1) {
+            this.setState({ status: 'dataError' });
+            console.log(cols);
+            alert(
+                'Mismatched column lengths, please select a correctly formatted file!'
+            );
+        } else if (!cols.has(2)) {
+            console.log(cols);
+            this.setState({ status: 'dataError' });
+            alert(
+                'Incorrect number of columns, please select a correctly formatted file!'
+            );
+        } else {
+            const clusterInfo = _.zip(...data);
+            this.setState({
+                status: 'ready',
+                newclusterInfo: {
+                    cellIDs: clusterInfo[0],
+                    clusterIDs: clusterInfo[1],
+                },
+            });
+        }
+    };
+
+    sendData = (e) => {
+        if (this.state.clusteringName == '') {
+            alert('You must enter a name for your clustering!');
+            return;
+        }
+        const retVal = confirm(
+            `Are you sure you want to add this new clustering?\n\n--> ${this.state.clusteringName} <--\n\nThis is a PERMANENT change!`
+        );
+        if (retVal == false) {
+            return;
+        }
+        BackendAPI.getConnection().then((gbc) => {
+            let query = {
+                loomFilePath: BackendAPI.getActiveLoom(),
+                orcidInfo: {
+                    orcidName: this.props.orcid_name,
+                    orcidID: this.props.orcid_id,
+                    orcidUUID: this.props.orcid_uuid,
+                },
+                clusterInfo: {
+                    ...{ clusteringName: this.state.clusteringName },
+                    ...this.state.newclusterInfo,
+                },
+            };
+            if (DEBUG) console.log('addNewClustering', query);
```

Comment:
```suggestion
            if (DEBUG) {
                console.debug('addNewClustering', query);
            }
```
Or better, a separate function